napari/utils/progress.py

```diff
@@ -138,19 +136,26 @@ def close(self):
         super().close()
 
     def close_pbar(self):
-        if not self.disable and self._pbar:
-            parent_widget = self._pbar.parent()
-            self._pbar.close()
-            self._pbar.deleteLater()
-            if isinstance(parent_widget, ProgressBarGroup):
-                pbar_children = [
-                    child
-                    for child in parent_widget.children()
-                    if isinstance(child, ProgressBar)
-                ]
-                if not any(child.isVisible() for child in pbar_children):
-                    parent_widget.close()
-            self._pbar = None
+        if self.disable or not self._pbar:
+            return
+
+        from napari._qt.widgets.qt_progress_bar import (
```

Comment:
another thought here: if you wanted to completely avoid this import here, you can make `progress` "evented"... add a close event, and hook all this closing logic up on the qt side in `get_pbar`

Reply:
I do like the idea of this being evented. Is that feasible? Would definitely deal with my concerns

Reply:
This was something @jni was also excited about, see #3032 for his suggestion that updates could also be linked to callbacks, which would allow the viewer to update a potentially separate `QProgressBar` and further decouple the `progress` object from the `Viewer`.

Reply:
I think it's definitely feasible, though I'm not sure how feasible it would then be to keep the integration with `tqdm`. Would have to investigate more closely.